source/client-side-encryption/client-side-encryption.rst

```diff
@@ -914,6 +994,11 @@ ClientEncryption
    class ClientEncryption {
       ClientEncryption(opts: ClientEncryptionOpts);
 
+      // The "Create Encrypted Collection" helper is a convenience function wrapping CreateCollection. It will
+      // create a collection with encrypted fields, automatically allocating and assigning new data encryption
+      // keys. Refer to "Create Encrypted Collection Helper"
+      createEncryptedCollection(collName: string, collOpts, kmsProvider: KMSProviderName, dkOpts: DataKeyOpts);
```

Comment:
as it was discussed offline:
>maybe another option may be passing not a clientEncription, but ClientEncryptionOptions  and create a client encryption internally each time?

Reply:
What's the benefit over taking the ClientEncryption object directly?

Reply:
it will work with Go since by this, as I understand, we avoid a cyclic physical dependency between components that prohibit using ClientEncryption directly as an argument. cc: @kevinAlbs

Reply:
It doesn't create a dependency cycle in Go, but what is the benefit of overloading CreateCollection to accept the options rather than creating the ClientEncryption from those options and using that?

In some cases (e.g. the C driver) we would still need to add an additional API to accept the options object since it cannot be fit into the existing `mongoc_database_create_collection` interface.

Reply:
> It doesn't create a dependency cycle in Go, but what is the benefit of overloading CreateCollection to accept the options rather than creating the ClientEncryption from those options and using that?

It may result in one less public method. Users wanting to create an encrypted collection only need to use CreateCollection. And CreateCollection already supports encryptedFields.

> In some cases (e.g. the C driver) we would still need to add an additional API to accept the options object since it cannot be fit into the existing mongoc_database_create_collection interface.

Good point. The current C options are not sufficient for accepting the keyVaultClient from ClientEncryptionOptions. C may be the only exception that would require additional API.

Is it OK to leave C as an exception, and have the specification specify in terms of CreateCollection?

Reply:
>what is the benefit of overloading CreateCollection 

the benefit is to have a single API to create encrypted collection and don't spread this logic across driver (ie it's better to have a single place for creating collection instead ClientEncryption and regular collection based approaches).

But there is one more downside with my suggestion that I didn't consider, not sure how likely, but in theory `ClientEncryptionOptions` may contain many options and some of them may be fully unrelated to creating collection, so it will be unclear what value should be provided there if we just want to create encrypted collection. 

Given the above (and the fact we probably don't want deprecating already existed approach or created some options hierarchy), maybe we can stick to the simplest approach suggested initially?

Reply:
If a user is intending to use an encrypted collection, they will likely need a `ClientEncryption` object anyway (at least for explicit encryption).

Even if it is an "overload" of the `CreateCollection` API, it will still be a distinct API that just happens to use the same name. Currently, the "CreateEncryptedCollection" is associated with the ClientEncryption object, but it could be migrated to be part of the Database APIs. It's mostly just a matter of parameter ordering.

Reply:
Good points. I am in favor of keeping the `CreateEncryptedCollection` API as is currently proposed.